cpp/src/arrow/ipc/json-simple.cc

```diff
@@ -319,6 +318,43 @@ class StringConverter final : public ConcreteConverter<StringConverter> {
   std::shared_ptr<BinaryBuilder> builder_;
 };
 
+// ------------------------------------------------------------------------
+// Converter for fixed-size binary arrays
+
+class FixedSizeBinaryConverter final
+    : public ConcreteConverter<FixedSizeBinaryConverter> {
+ public:
+  explicit FixedSizeBinaryConverter(const std::shared_ptr<DataType>& type) {
+    this->type_ = type;
+    builder_ = std::make_shared<FixedSizeBinaryBuilder>(type, default_memory_pool());
+  }
+
+  Status AppendNull() override { return builder_->AppendNull(); }
+
+  Status AppendValue(const rj::Value& json_obj) override {
```

Comment:
I'd refactor this function with a simple switch (if possible?) on Json type and hide the string_view parsing in a static inline function.

```c++
switch(json_type) {
  case Json::NULL: return AppendNull();
  case Json::STRING: {
    util::string_view view;
    ARROW_RETURN_NOT_OK(JsonStringToViewOfFixedSize(json_obj, fixed_size, &view));
    return builder_->Append(view);
  }
  default: return Status::Invalid(...);
}
```

This is a style preference, but I think it improves reading since there's less control flow involved.

Reply:
The current idiom (a chain of `if`s) is used elsewhere in the module, though. So we'd have to change other occurrences as well. IMHO there's not much benefit.

Reply:
I think it's OK to leave the code as is since it's similar to the rest of the file. If future refactoring is needed then whoever is working on it is free to do this code scrubbing